Added Flaky decorator to Chrome flaky tests

raeeschachar opened the pull request.

raeeschachar commented:
jenkins run bokchoy

raeeschachar commented:
jenkins run all

raeeschachar commented:
jenkins run all

raeeschachar commented:
jenkins run python

raeeschachar commented:
jenkins run python

raeeschachar commented:
jenkins run python

raeeschachar commented:
@benpatterson @jzoldak @Muddasser 
Please review. 
Should I also add Jira tickets for these?

raeeschachar commented:
jenkins run bokchoy

raeeschachar commented:
@benpatterson @jzoldak 
[latest run using this PR] (https://build.testeng.edx.org/view/All/job/edx-platform-bok-choy-custom/109/testReport/)
Studying Flaky and Skip I think using '@skip_if_browser('chrome')' would be a better solution then this. 
Any suggestions?

benpatterson commented:
@raeeschachar will it work with `@flaky` on chrome? If it'll work with a re-run, I'd rather do that than skip it altogether for chrome. However, if it's _always_ broken on chrome, then I'd want to hear more about the issue....

raeeschachar commented:
@benpatterson 
1- Seeing the test results, on Jenkins with this PR, it will pass but sporadically. Not sure how many times we will have to run it. 
2- This test 
'common.test.acceptance.tests.studio.test_studio_asset.AssetIndexTest.test_page_existence' is the issue
The following 3 tests are dependent on the above test:
'common.test.acceptance.tests.studio.test_studio_asset.AssetIndexTest.test_type_filter_exists'
'common.test.acceptance.tests.studio.test_studio_asset.AssetIndexTest.test_filter_results'
'common.test.acceptance.tests.studio.test_studio_general.CoursePagesTest.test_page_existence' 
All of these call AssetIndexTest.test_page_existence(i.e the first test) and fail there. 

When I run a Jenkins job against this PR, sometimes 2 fail sometimes 3 fail and all sporadically. 
Comparisons [one] (https://build.testeng.edx.org/view/All/job/edx-platform-bok-choy-custom/105/testReport/) and [two] (https://build.testeng.edx.org/view/All/job/edx-platform-bok-choy-custom/109/testReport/)

jzoldak commented:
Can we figure out a way to combine the two approaches? That is, to apply the flaky decorator only in the case that the test is running with chrome as the browser.

raeeschachar commented:
@benpatterson @jzoldak @Muddasser 
Added skip if chrome to:
common.test.acceptance.tests.studio.test_studio_asset.py
Added flaky to: 
common.test.acceptance.tests.studio.test_studio_settings_certificates.test_can_create_and_edit_signatories_of_certficate

I still need to test these running a jenkins job.

raeeschachar commented:
jenkins run bokchoy

raeeschachar commented:
@benpatterson @jzoldak @Muddasser 
https://build.testeng.edx.org/view/All/job/edx-platform-bok-choy-custom/114/testReport/
against this PR

benpatterson commented:
Is your branch recently rebased? I'm pretty sure the mathjax flakiness has been fixed.

Anyway I am good to move forward with this, with some tickets to take care of later, as you originally suggested.

:+1:

jzoldak commented:
:+1: 
One thing that is strange to me so I'm curious about the answer (but I don't want to hold up this PR over it) is that it takes 3.5 seconds for each of the asset tests that it is skipping. 

https://build.testeng.edx.org/view/All/job/edx-platform-bok-choy-custom/114/testReport/common.test.acceptance.tests.studio.test_studio_asset/AssetIndexTest/

@raeeschachar do you happen to know what it's doing during this time?

Another comment I have, at looking at those asset tests, is that test_page_existence is more like a unit test than an acceptance test. Its outcome is covered in the other 2 tests in the AssetIndexTest, and as a discreet test it is unnecessary. @raeeschachar could you please simply remove it, either in this PR or another one?

raeeschachar commented:
@jzoldak not sure why it is taking 3.5 seconds  or what it is doing in that time 
Other comments addressed. Good to go?

jzoldak commented:
@raeeschachar yes, good to merge! :rocket: :+1:
